packages/plugins/response-cache-cloudflare-kv/src/invalidate.ts

```diff
@@ -0,0 +1,44 @@
+import type { CacheEntityRecord } from '@envelop/response-cache';
+import type { KvCacheConfig } from '.';
+import { buildEntityKey } from './cache-key';
+
+export async function invalidate(
+  entities: Iterable<CacheEntityRecord>,
+  config: KvCacheConfig,
+): Promise<void> {
+  const kvPromises: Promise<unknown>[] = []; // Collecting all the KV operations so we can await them all at once
+
+  for (const entity of entities) {
+    const entityKey = buildEntityKey(entity.typename, entity.id, config.keyPrefix);
+
+    for await (const kvKey of _getAllKvKeysForPrefix(entityKey, config)) {
+      if (kvKey.metadata?.operationKey) {
+        kvPromises.push(config.KV.delete(kvKey.metadata?.operationKey));
+        kvPromises.push(config.KV.delete(kvKey.name));
+      }
+    }
+  }
+
+  await Promise.allSettled(kvPromises);
```

Comment:
Perhaps we could even do this in an async function to allow parallelisation ?

```suggestion
  await Promise.allSettled(entities.map(async entity => {
    const kvPromises: Promise<unknown>[] = []; // Collecting all the KV operations so we can await them all at once
    const entityKey = buildEntityKey(entity.typename, entity.id, config.keyPrefix);

    for await (const kvKey of _getAllKvKeysForPrefix(entityKey, config)) {
      if (kvKey.metadata?.operationKey) {
        kvPromises.push(config.KV.delete(kvKey.metadata?.operationKey));
        kvPromises.push(config.KV.delete(kvKey.name));
      }
    }
    
    return Promise.allSettled(kvPromises);
  }));
```

Reply:
That's a great catch!